packages/ui/src/accounts/model/toBalances.ts

```diff
@@ -0,0 +1,32 @@
+import { DeriveBalancesAll } from '@polkadot/api-derive/types'
+import { LockIdentifier } from '@polkadot/types/interfaces'
+import BN from 'bn.js'
+
+import { lockTypes } from '@/accounts/model/lockTypes'
+import { Balances } from '@/accounts/types'
+import { capitalizeFirstLetter } from '@/common/helpers'
+
+function lockLookup(id: LockIdentifier) {
+  return lockTypes[id.toHex()] || capitalizeFirstLetter(<string>id.toHuman()).trim()
+}
+
+export function toBalances(balances: DeriveBalancesAll): Balances {
+  const { lockedBalance, availableBalance, lockedBreakdown, freeBalance } = balances
+
+  const locks = lockedBreakdown.map((lock) => ({
+    amount: lock.amount,
+    type: lockLookup(lock.id),
+  }))
+
+  const transferable = availableBalance.toBn()
+  const locked = lockedBalance.toBn()
+  const total = freeBalance.toBn() // Should be .add(reservedBalance) ?
```

Comment:
I think `const total = freeBalance.add(lockedBalance)` according to issue description

Reply:
I've just found this comprehensive answer: [https://stackoverflow.com/a/56418101/2273046.](https://stackoverflow.com/a/56418101/2273046.) 

I'm still not 100% if Joystream will allow reserving a balance.

So,

*   total = free + reserved
*   transferable = free - locked (looks like duplicates `available`)